contracts/test/integration/L2PlusSubnet.t.sol

```diff
@@ -0,0 +1,857 @@
+import {GatewayDiamond} from "../../contracts/GatewayDiamond.sol";
+import {SubnetActorDiamond} from "../../contracts/SubnetActorDiamond.sol";
+import {SubnetActorManagerFacet} from "../../contracts/subnet/SubnetActorManagerFacet.sol";
+import {SubnetActorCheckpointingFacet} from "../../contracts/subnet/SubnetActorCheckpointingFacet.sol";
+import {GatewayGetterFacet} from "../../contracts/gateway/GatewayGetterFacet.sol";
+import {LibGateway} from "../../contracts/lib/LibGateway.sol";
+import {TopDownFinalityFacet} from "../../contracts/gateway/router/TopDownFinalityFacet.sol";
+import {CheckpointingFacet} from "../../contracts/gateway/router/CheckpointingFacet.sol";
+import {XnetMessagingFacet} from "../../contracts/gateway/router/XnetMessagingFacet.sol";
+import {DiamondCutFacet} from "../../contracts/diamond/DiamondCutFacet.sol";
+import {GatewayMessengerFacet} from "../../contracts/gateway/GatewayMessengerFacet.sol";
+import {IntegrationTestBase, RootSubnetDefinition, TestSubnetDefinition} from "../IntegrationTestBase.sol";
+import {TestUtils, MockIpcContract, MockIpcContractPayable, MockIpcContractResult} from "../helpers/TestUtils.sol";
+import {FilAddress} from "fevmate/contracts/utils/FilAddress.sol";
+import {MerkleTreeHelper} from "../helpers/MerkleTreeHelper.sol";
+import {GatewayFacetsHelper} from "../helpers/GatewayFacetsHelper.sol";
+import {SubnetActorFacetsHelper} from "../helpers/SubnetActorFacetsHelper.sol";
+
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {ERC20PresetFixedSupply} from "../helpers/ERC20PresetFixedSupply.sol";
+
+import "forge-std/console.sol";
+
+contract L2PlusSubnetTest is Test, IntegrationTestBase {
+    using SubnetIDHelper for SubnetID;
+    using CrossMsgHelper for IpcEnvelope;
+    using GatewayFacetsHelper for GatewayDiamond;
+    using SubnetActorFacetsHelper for SubnetActorDiamond;
+    using AssetHelper for Asset;
+
+    RootSubnetDefinition public rootNetwork;
+    // native subnets
+    TestSubnetDefinition public nativeSubnet;
+    TestSubnetDefinition[] public nativeL3Subnets;
+
+    // token subnets
+    IERC20 public token;
+    TestSubnetDefinition public tokenSubnet;
+    TestSubnetDefinition[] public nativeL3SubnetsWithTokenParent;
+    IERC20 public tokenL3;
+    TestSubnetDefinition[] public tokenL3SubnetsWithTokenParent;
+
+    function setUp() public override {
+        SubnetID memory rootNetworkName = SubnetID({root: ROOTNET_CHAINID, route: new address[](0)});
+        require(rootNetworkName.isRoot(), "not root");
+
+        GatewayDiamond rootGateway = createGatewayDiamond(gatewayParams(rootNetworkName));
+
+        rootNetwork = RootSubnetDefinition({
+            gateway: rootGateway,
+            gatewayAddr: address(rootGateway),
+            id: rootNetworkName
+        });
+
+        address[] memory nativeSubnetPath;
+        nativeSubnet = createNativeSubnet(nativeSubnetPath, rootNetwork.gatewayAddr, rootNetwork.id);
```

Comment:
maybe call this `nativeL2Subnet`?

Reply:
@raulk Solved by using the parent route instead.